solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -2131,12 +2128,7 @@ public LeaderElector getOverseerElector() {
   static String generateNodeName(final String hostName,
                                  final String hostPort,
                                  final String hostContext) {
-    try {
-      return hostName + ':' + hostPort + '_' +
-          URLEncoder.encode(trimLeadingAndTrailingSlashes(hostContext), "UTF-8");
-    } catch (UnsupportedEncodingException e) {
-      throw new Error("JVM Does not seem to support UTF-8", e);
-    }
+    return hostName + ':' + hostPort + '_' + URLEncoder.encode(trimLeadingAndTrailingSlashes(hostContext), StandardCharsets.UTF_8);
```

Comment:
Java8 doesn't have `URLEncoder.encode(String, Charset)`. Also, it'd through the exception on 8x.
I like it this way, just saying that you might have to remember that when merging.